### PR TITLE
add button to export families as a csv

### LINKED ISSUE #252: Add export as CSV functionality for all resources
This needs to be fleshed out and broken out into different issues, but for now Ill list the page where we will want this...

- [ ] Pickup sheets
- [ ] Families
- [ ] Children

### PR DESCRIPTION
**Pull request by gabrielbaldao**
Resolves #252  

### Type of change

* New feature (non-breaking change which adds functionality)
### How Has This Been Tested?

With a spec testing content type request, and i created a family_controller_spec to start with new tests about what was created.

### Screenshots

Before
before_true

Now
now

Csv file
csv

### CODE REVIEW

File: spec/controllers/families_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+describe FamiliesController, type: :controller do
+  let(:partner) { create :partner, name: 'new partner' }
+  let(:user) { create :user, partner: partner }
+  let!(:family) { create :family, partner: partner }
+
+  before do
+    sign_in user
+  end
+  describe "GET #inde" do
```
Comment (renatamarques97) [inline spec/controllers/families_controller_spec.rb]: don't forget the 'x' here

File: spec/controllers/families_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    before do
+      user.partner = partner
+    end
+    describe "when responding csv" do
```
Comment (renatamarques97) [inline spec/controllers/families_controller_spec.rb]: replace "describe" with "context"

File: spec/controllers/families_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  let(:partner) { create :partner, name: 'new partner' }
+  let(:user) { create :user, partner: partner }
+  let!(:family) { create :family, partner: partner }
+
+  before do
+    sign_in user
+  end
+  describe "GET #inde" do
+    before do
+      user.partner = partner
```
Comment (renatamarques97) [inline spec/controllers/families_controller_spec.rb]: you don't need two "before do", one is enough

**Comment by renatamarques97** [comment]
rubocop is failing, please take a look.

**Comment by benreyn** [comment]
Thanks for the contribution @gabrielbaldao. And thank you for giving it a review @renatamarques97 ! Really appreciate it!

File: app/controllers/families_controller.rb [hunk 4]
```diff
@@ -1,8 +1,15 @@
+require "csv"
 class FamiliesController < ApplicationController
   before_action :authenticate_user!
 
   def index
     @families = current_partner.families.order(:guardian_last_name)
+    respond_to do |format|
+      format.html
+      format.csv do
+        render(csv: @families.map.map(&:to_csv))
```
Comment (benreyn) [inline app/controllers/families_controller.rb]: Pretty sure you can get away with only one call to `map` here.

**Pull request merged**